pkg/util/test/util.go

```diff
@@ -2951,6 +2956,172 @@ func ValidatePrometheus(pxImageList map[string]string, cluster *corev1.StorageCl
 	return nil
 }
 
+func ValidateGrafana(cluster *corev1.StorageCluster) error {
+	opVersion, err := GetPxOperatorVersion()
+	if err != nil {
+		return err
+	}
+	if opVersion.LessThan(opVer23_8) {
+		logrus.Infof("Skipping grafana validation for operation version: [%s]", opVersion.String())
+		return nil
+	}
+
+	shouldBeInstalled := cluster.Spec.Monitoring != nil && cluster.Spec.Monitoring.Grafana != nil &&
+		cluster.Spec.Monitoring.Grafana.Enabled && cluster.Spec.Monitoring.Prometheus != nil &&
+		cluster.Spec.Monitoring.Prometheus.Enabled
+	image := cluster.Status.DesiredImages.Grafana
```

Comment:
I would recommend to follow how we validate all expected images, from `pxImageList`, because that is what we want to be installed, because `DesiredImages` is what PX Operator decided to install, but it doesn't necessary match which what it should be installing. The correct images should come versions manifest, wether its `px-versions` configmap or URL endpoint

Reply:
Just pass the `pxImageList` to the function, find your Grafana image there, which should be `pxImageList["grafana"]`, this should give you the value of it like so `grafana: docker.io/grafana/grafana:7.5.17`, then use the `validateContainerImageInsidePods()` function, because it will correctly trim the image to the correct URN, incase of custom registry or anything like that